### PR TITLE
recover example

### PR DESCRIPTION
**Pull request by kilaka**

### CODE REVIEW

File: examples/recover/recover.go
Comment (eliben) [inline examples/recover/recover.go:11]: There aren't many comments in this sample. Is this intentional?

File: examples/recover/recover.go
Comment (eliben) [inline examples/recover/recover.go:19]: Not sure if having two almost identical functions for builtin and custom panics is really useful here.

All things being equal, the sample should be as small as possible. The goal here is to remember (or copy-paste) the proper invocation of `recover` inside a `defer`, not explore different ways in which Go can panic.

A simple "custom" example should suffice. The panic example has a panic, the recover example should just recover from the same panic.

**Review by eliben — CHANGES_REQUESTED** [review]

File: examples/recover/recover.go
Comment (kilaka) [reply examples/recover/recover.go:12]: Added some comments in 47eed5ec17c1925520196d718befa82e69f5ccbc.
Coming from Java's "exceptions world", the recover seems a bit trivial, so I'm not sure what more info should I add.

File: examples/recover/recover.go
Comment (kilaka) [reply examples/recover/recover.go]: Fixed as you requested in 47eed5ec17c1925520196d718befa82e69f5ccbc.
Thanks for the feedback.

Comment (kilaka) [reply examples/recover/recover.go]: But I didn't use the same panic, as I think it's important to see that the panic might happen in a "lower" function call.
If it's critical I can change to align with the panic example. Or perhaps align the panic example?
Whatever you prefer...

**Comment by kilaka** [comment]
Sorry guys but I'm still stuck with two problems which I hope you know how to fix quickly:
1. Some strange auto-generated chars, which changed from `\x3D` to `\u003D`, causing a lot of files to look as if they were changed, but actually were not. (Perhaps related to working on a Linux laptop?).
2. Misalignment between explanation/comments and code, specifically the comment above the recover.

**Comment by eliben** [comment]
IMHO the example should be even simpler, something like (with appropriate comments):

```
func mayPanic() {
	panic("a problem")
}

func main() {
	defer func() {
		if r := recover(); r != nil {
			fmt.Println("Recovered. Error:\n", r)
		}
	}()

	mayPanic()
	fmt.Println("After mayPanic()")
}
```

The main idea is not necessarily to teach all the ways in which panic/recover works, but to provide a small copy-pasteable example folks can use if they don't exactly remember the syntax/mechanics. I agree that it's useful to place the potential panic in a function, but the recover can be in `main`, and there's no need for any additional logic that's not specifically required for the panic->recover chain to work.

But this is just my opinion :-) I'll leave it to @mmcgrana to decide.

**Comment by eliben** [comment]
For backlinking, this PR is related to issue #46

**Comment by eliben** [comment]
@kilaka do you intend to keep working on this?

**Comment by kilaka** [comment]
@eliben , not soon. But I thought we're waiting for @mmcgrana 's feedback...

> @kilaka do you intend to keep working on this?

**Comment by mmcgrana** [comment]
@kilaka I think your example is in the right direction. I agree the example should be as simple as possible,  @eliben's suggestion there seems plausible. I'd also want some brief comments about when/why one uses `recover`, especially since it is IIUC an unusual tool for application-level programers. See e.g. the Atomic Counter vs. Mutex vs. Stateful Goroutine pages for examples of lightweight when/why discussion. I've never used `recover` in my entire life as far as I know, so it's not easy for me to suggest particular comments, sorry 😬

**Comment by eliben** [comment]
Thanks for discussion. I ended up adding a simplified example (see linked commit). Please open an issue/PR if you have questions or suggestions about the new example.

**Pull request closed without merging**
